### ISSUE TITLE
Jet prefix visibility in 4.0

### ISSUE THREAD

**Issue by neilstevenson** [hazelcast/hazelcast-jet#2085]
This https://github.com/hazelcast/hazelcast-jet/issues/583 made the prefix (`__jet`) visible in 3.*.

In 4.0, this is now part of an _impl_ class https://github.com/hazelcast/hazelcast-jet/blob/master/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JobRepository.java#L91

There should be an API getter to find this, as an _impl_ class may change

I'd like to be able to do
```
instance.getDistributedObjects()
                    .stream()
                    .filter(distributedObject -> distributedObject instanceof IMap)
                    .map(distributedObject -> distributedObject.getName())
                    .filter(name -> !name.startsWith(JobRepository.INTERNAL_JET_OBJECTS_PREFIX))
                    .collect(Collectors.toCollection(TreeSet::new));
```
solely through the API

**Comment by Sunil-Jain** [comment]
Hi @Holmistr ,

I would like to work on this issue.

**Comment by mtopolnik** [comment]
My suggestion is to move the constant to `JetConfig`.

**Comment by neilstevenson** [comment]
A method rather than a constant may be preferable.

The job prefix "`__`" and the product "`jet`" form the prefix for the map. 

IMDG does yet create system objects, but may in the future. So the prefix part "`__`" should really come from IMDG.

And, as a very long way off, the end user may wish to be able to override the prefix part"`__`" if this somehow clashes with their own map naming.

**Comment by mtopolnik** [comment]
I think we should stick to the KISS and YAGNI principles and keep it just a constant. If it becomes insufficient in the future, we can change  it then.

**Comment by Sunil-Jain** [comment]
I also agree with @mtopolnik . We should stick with KISS and YAGNI principles and keep it just a constant.

**Comment by bfeldkamp3** [comment]
@Sunil-Jain are you still working this? If not, I'd like to take it over 😄.
